tee-worker/core/parentchain/indirect-calls-executor/src/executor/litentry/batch_all.rs

```diff
@@ -0,0 +1,422 @@
+						let xt = ParentchainUncheckedExtrinsic {
+							function: call,
+							signature: extrinsic.signature.clone(),
+						};
+						create_identity.execute(context, xt)?;
+					}
+				},
+				itp_types::SupportedCall::VerifyIdentity(call_index, Some(params)) => {
+					let verify_identity = VerifyIdentity { block_number: self.block_number };
+					let call = (*call_index, params.clone());
+					if Executor::<
+						ShieldingKeyRepository,
+						StfEnclaveSigner,
+						TopPoolAuthor,
+						NodeMetadataProvider,
+					>::is_target_call(
+						&verify_identity, &call, context.node_meta_data_provider.as_ref()
+					) {
+						let xt = ParentchainUncheckedExtrinsic {
+							function: call,
+							signature: extrinsic.signature.clone(),
+						};
+						verify_identity.execute(context, xt)?;
+					}
+				},
+				itp_types::SupportedCall::RemoveIdentity(call_index, Some(params)) => {
+					let remove_identity = RemoveIdentity {};
+					let call = (*call_index, params.clone());
+					if Executor::<
+						ShieldingKeyRepository,
+						StfEnclaveSigner,
+						TopPoolAuthor,
+						NodeMetadataProvider,
+					>::is_target_call(
+						&remove_identity, &call, context.node_meta_data_provider.as_ref()
+					) {
+						let xt = ParentchainUncheckedExtrinsic {
+							function: call,
+							signature: extrinsic.signature.clone(),
+						};
+						remove_identity.execute(context, xt)?;
+					}
+				},
+				_ => {
+					log::warn!("Unimplemented. Call: {:?}", call);
+				},
+			};
+		}
+		Ok(())
+	}
+}
+
+impl<ShieldingKeyRepository, StfEnclaveSigner, TopPoolAuthor, NodeMetadataProvider>
+	Executor<ShieldingKeyRepository, StfEnclaveSigner, TopPoolAuthor, NodeMetadataProvider> for BatchAll
+where
+	ShieldingKeyRepository: AccessKey,
+	<ShieldingKeyRepository as AccessKey>::KeyType: ShieldingCryptoDecrypt<Error = itp_sgx_crypto::Error>
+		+ ShieldingCryptoEncrypt<Error = itp_sgx_crypto::Error>,
+	StfEnclaveSigner: StfEnclaveSigning,
+	TopPoolAuthor: AuthorApi<H256, H256> + Send + Sync + 'static,
+	NodeMetadataProvider: AccessNodeMetadata,
+	NodeMetadataProvider::MetadataType:
+		IMPCallIndexes + TeerexCallIndexes + VCMPCallIndexes + UTILCallIndexes,
+{
+	type Call = BatchAllFn;
+
+	fn decode(
+		&self,
+		context: &IndirectCallsExecutor<
+			ShieldingKeyRepository,
+			StfEnclaveSigner,
+			TopPoolAuthor,
+			NodeMetadataProvider,
+		>,
+		input: &mut &[u8],
+	) -> Result<itp_types::extrinsics::ParentchainUncheckedExtrinsicWithStatus<Self::Call>> {
+		// This is a little more complicated than usual since the binary format must be compatible
+		// with substrate's generic `Vec<u8>` type. Basically this just means accepting that there
+		// will be a prefix of vector length (we don't need
+		// to use this).
+		// OpaqueExtrinsicWithStatus & OpaqueExtrinsic together have two leyers encode with vec.
+		let _length_do_not_remove_me_see_above: Vec<()> = Decode::decode(input)?;
+		let _length_do_not_remove_me_see_above: Vec<()> = Decode::decode(input)?;
```

Comment:
Why do we need twice?

Reply:
Just above the code, I left comment there. Because OpaqueExtrinsicWithStatus & OpaqueExtrinsic together have two layers encode with vec.